trillian/migrillian/core/controller.go

```diff
@@ -53,14 +77,23 @@ type Controller struct {
 	ctClient *client.LogClient
 	plClient *PreorderedLogClient
 	ef       election.Factory
+
+	mon   metrics
```

Comment:
Not sure that metrics belong in options. They're owned by the server usually. Also name "mon" is somewhat unclear. 

Is this to make them per mirror? That probably won't work as they have to have unique names. This should probably work like CTFE where there's a single set of metrics using labels to distinguish the logs.

Reply:
Metrics belong to the `Controller` (GitHub collapsed the lines between `Options` and `Controller`, that might've been misleading to you). The "mon" name stands for "monitoring". I think it's a pretty common shortcut (and shortcuts in Go are okay), but I renamed it to "metrics".

Regarding the labels, see `newMetrics`: there is a `treeID` label which I have in each metric.

Reply:
Is there one controller or one controller per tree?

Reply:
Currently it's one per tree (although Migrillian doesn't support running multiple), but I'm planning to also have a multi-tenant SuperController above them. So things like election factory and metrics need to be stored in SuperController once it's introduced. I can leave a TODO.